bash-preexec.sh

```diff
@@ -83,10 +86,17 @@ __bp_interactive_mode() {
 # This function is installed as part of the PROMPT_COMMAND.
 # It will invoke any functions defined in the precmd_functions array.
 __bp_precmd_invoke_cmd() {
-
     # Save the returned value from our last command
```

Comment:
Since this is problem is quite unintuitive, I think putting a comment like

    # Save the returned value from our last command. Note: this MUST be the
    # first thing done in this function.

would help avoid it resurfacing in the future.